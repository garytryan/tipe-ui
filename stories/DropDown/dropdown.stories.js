import { storiesOf } from '@storybook/vue'

import TipeDropDown from '../../src/components/DropDown'
import TipeButton from '@/components/Button'

const style = () => ({
  width: '200px'
})

storiesOf('DropDown', module)
  .add('default', () => ({
    components: { TipeDropDown, TipeButton },
    computed: { style },
    template: `
      <tipe-drop-down name="firstName">
        <div slot="control"><tipe-button size="small">Click Me!</tipe-button></div>
        <div slot="panel">
          <ul>
            <li>option</li>
            <li>option</li>
            <li>option</li>
          </ul>
        </div>
      </tipe-drop-down>`
  }))
  .add('disabled', () => ({
    components: { TipeDropDown, TipeButton },
    computed: { style },
    template: `
      <tipe-drop-down name="firstName" disabled="true">
        <div slot="control"><tipe-button size="small" disabled>Click Me!</tipe-button></div>
        <div slot="panel">
          <ul>
            <li>option</li>
            <li>option</li>
            <li>option</li>
          </ul>
        </div>
      </tipe-drop-down>`
  }))
