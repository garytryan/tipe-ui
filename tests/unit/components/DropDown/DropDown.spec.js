import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import DropDown from '@/components/DropDown'

describe('DropDown', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(DropDown)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(DropDown)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeDropDown')
    })
  })
})
