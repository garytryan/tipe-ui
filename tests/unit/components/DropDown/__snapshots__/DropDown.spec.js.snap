// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeDropDown">
  <div class="control-container"></div>
  <!---->
  <!---->
</div>
`;
